File: JavaDemo/app/src/main/java/com/example/javademo/PostAdapter.java
package com.example.javademo;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import java.lang.reflect.Array;
import java.util.ArrayList;
import java.util.List;

public class PostAdapter extends RecyclerView.Adapter<PostAdapter.ViewHolder> {
    private ArrayList<Post> posts = new ArrayList<>();

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.post, parent, false);
        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        Post post = posts.get(position);
        holder.bindData(post);
    }

    @Override
    public int getItemCount() {
        return posts.size();
    }

    public void setData(List<Post> postList) {
        this.posts.clear();
        this.posts.addAll(postList);
        this.notifyDataSetChanged();
    }

    public static class ViewHolder extends RecyclerView.ViewHolder {
        private final TextView id;
        private final TextView userId;
        private final TextView title;
        private final TextView content;

        public ViewHolder(@NonNull View itemView) {
            super(itemView);
            id = itemView.findViewById(R.id.id_detail);
            userId = itemView.findViewById(R.id.user_id);
            title = itemView.findViewById(R.id.post_title);
            content = itemView.findViewById(R.id.post_content);
        }

        public void bindData(Post post) {
            id.setText("Id: " + post.getId());
            userId.setText("User Id: " + post.getUserId());
            title.setText(post.getTitle());
            content.setText(post.getBody());
        }
    }
}

File: KotlinDemo/app/src/main/java/com/example/kotlindemo/PostAdapter.kt
package com.example.kotlindemo

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import java.util.ArrayList

class PostAdapter : RecyclerView.Adapter<PostAdapter.ViewHolder>() {
    private val posts = ArrayList<Post>()
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.post, parent, false)
        return ViewHolder(view)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val post = posts[position]
        holder.bindData(post)
    }

    override fun getItemCount(): Int {
        return posts.size
    }

    fun setData(postList: List<Post>?) {
        posts.clear()
        posts.addAll(postList!!)
        notifyDataSetChanged()
    }

    class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        private val id: TextView
        private val userId: TextView
        private val title: TextView
        private val content: TextView
        fun bindData(post: Post) {
            id.text = "Id: ${post.id}"
            userId.text = "User Id: ${post.userId}"
            title.text = post.title
            content.text = post.body
        }

        init {
            id = itemView.findViewById(R.id.id_detail)
            userId = itemView.findViewById(R.id.user_id)
            title = itemView.findViewById(R.id.post_title)
            content = itemView.findViewById(R.id.post_content)
        }
    }
}